### PR TITLE
scaffolding for plugin architecture 

### PR DESCRIPTION
**Pull request by avaldebe**
Provide some readers via entry_points

### CODE REVIEW

File: pyaerocom/io/readungridded.py [hunk 1]
```diff
@@ -52,14 +56,15 @@ class ReadUngridded:
         ReadAeronetSunV3,
         ReadEarlinet,
         ReadEbas,
-        ReadGAW,
         ReadAasEtal,
-        ReadGhost,
         ReadAirNow,
```
Comment (lewisblake) [inline pyaerocom/io/readungridded.py:60]: It's not clear to me why only GAW, Ghost, and MSCW CTM have their own plugins. Is there something special about these or do we just need to begin moving the other readers into this format later?

File: pyproject.toml [hunk 2]
```diff
@@ -1,6 +1,99 @@
...
+    "Development Status :: 3 - Alpha",
+    "Intended Audience :: Science/Research",
+    "Intended Audience :: Education",
+    "Topic :: Scientific/Engineering :: Atmospheric Science",
+]
+requires-python = ">=3.8"
+dependencies = [
+    'importlib-metadata>=3.6; python_version < "3.10"',
+    'tomli>=2.0.1; python_version < "3.11"',
+    # scitools-iris>=3.1.0,!=3.2.*
```
Comment (lewisblake) [inline pyproject.toml:26]: Consider removing these commented out lines. Looks like some are covered by `[project.optional-dependencies]` below

**Review by lewisblake — COMMENTED** [review]
Little things. I think this architecture makes a lot more sense and helps split things into more manageable pieces. Godt jobbet

File: pyaerocom/io/readungridded.py [hunk 3]
```diff
@@ -52,14 +56,15 @@ class ReadUngridded:
...
         ReadMarcoPolo,
         ReadEEAAQEREP,
         ReadEEAAQEREP_V2,
     ]
+    SUPPORTED_READERS.extend(
+        ep.load() for ep in metadata.entry_points(group="pyaerocom.ungridded")
+    )
```
[the remark this answers is not in the document]
Comment (avaldebe) [reply pyaerocom/io/readungridded.py:67]: A broken module would raise a `SyntaxError`, which should cause the test suite to fail.

A more subtle fail would be a missing dependency, which would raise a `ModuleNotFoundError`.
We could try/except this exception and issue a `warnings.warn`, but any evaluation relying on the missing reader would crash at a later point.
I think that it is best to let it fail at this point.

File: pyaerocom/io/readungridded.py (hunk 1 above)
Comment (avaldebe) [reply pyaerocom/io/readungridded.py:60]: These were easy to port. Other readers are more complicated.
We can port more readers on subsequent PRs.

File: pyproject.toml (hunk 2 above)
Comment (avaldebe) [reply pyproject.toml:26]: god point, will do

File: pyaerocom/plugins/ghost/additional_variables.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    data : list
+        list of modified :class:`StationData` objects that include computed
+        mass concentrations in addition to VMR data.
+
+    """
+    for stat in data:
+        vmrdata = stat[vmrvar]
+        meta = stat["meta"]
+        p = meta["network_provided_volume_standard_pressure"]
+        T = meta["network_provided_volume_standard_temperature"]
```
[the remark this answers is not in the document]
Comment (avaldebe) [reply pyaerocom/plugins/ghost/additional_variables.py:40]: So far, I only moved the code around and renamed some modules.
I will rename this variables following your suggestion

**Review by jgriesfeller — APPROVED** [review]
Looks good to me

**Pull request merged**